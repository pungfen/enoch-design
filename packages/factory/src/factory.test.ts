import { factory } from '.'

describe('factory', () => {
  const f = factory({
    props: {
      type: Number
    },

    setup: {
      form: {
        ajax: {
          action: 'GET /test',
          data: 'object'
        }
      }
    },

    onMounted() {
      this.type
    }
  })

  it('Math.sqrt()', () => {
    expect(Math.sqrt(4)).toBe(2)
    expect(Math.sqrt(144)).toBe(12)
    expect(Math.sqrt(2)).toBe(Math.SQRT2)

    const result = Math.sqrt(16)
    expect(result).toMatchSnapshot()
    expect(result).toMatchInlineSnapshot('4')
  })
})

describe.skip('fp', () => {
  it('skipped test', () => {
      assert.equal(Math.sqrt(4), 10)
  })
})

describe.todo('unimplemented suite')

describe('suite', () => {
  it.todo('unimplemented test')
})

