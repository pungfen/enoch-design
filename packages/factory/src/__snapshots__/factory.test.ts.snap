// Vitest Snapshot v1

exports[`factory > Math.sqrt() 1`] = `4`;
